applications/StructuralMechanicsApplication/tests/cpp_tests/test_mpc_plate_tension.cpp

```diff
@@ -0,0 +1,178 @@
+typedef UblasSpace<double, CompressedMatrix, Vector> SparseSpaceType;
+typedef UblasSpace<double, Matrix, Vector> LocalSpaceType;
+typedef LinearSolver<SparseSpaceType,LocalSpaceType> LinearSolverType;
+typedef SkylineLUFactorizationSolver<SparseSpaceType,  LocalSpaceType > SkylineLUFactorizationSolverType;
+typedef BuilderAndSolver< SparseSpaceType, LocalSpaceType, LinearSolverType > BuilderAndSolverType;
+typedef ResidualBasedBlockBuilderAndSolver< SparseSpaceType, LocalSpaceType, LinearSolverType > ResidualBasedBlockBuilderAndSolverType;
+typedef Scheme< SparseSpaceType, LocalSpaceType >  SchemeType;
+typedef ResidualBasedIncrementalUpdateStaticScheme< SparseSpaceType, LocalSpaceType> ResidualBasedIncrementalUpdateStaticSchemeType;
+typedef SolvingStrategy<SparseSpaceType, LocalSpaceType, LinearSolverType> SolvingStrategyType;
+typedef ResidualCriteria< SparseSpaceType, LocalSpaceType > ResidualCriteriaType;
+typedef ResidualBasedLinearStrategy< SparseSpaceType, LocalSpaceType, LinearSolverType > ResidualBasedLinearStrategyType;
+
+KRATOS_TEST_CASE_IN_SUITE(PatchTestMPCPlateTension, KratosStructuralMechanicsFastSuite)
+{
+    Model current_model;
+    ModelPart& model_part = current_model.CreateModelPart("Main");
+    model_part.SetBufferSize(2);
+    model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+    model_part.AddNodalSolutionStepVariable(REACTION);
+    //Material properties
+    auto p_Properties = model_part.CreateNewProperties(0);
+    p_Properties->SetValue(YOUNG_MODULUS, 210e3);
+    p_Properties->SetValue(POISSON_RATIO, 0.3);
+    p_Properties->SetValue(DENSITY, 1.0);
+    p_Properties->SetValue(THICKNESS, 1.0);
+    auto p_constitutive_law = KratosComponents<ConstitutiveLaw>::Get("LinearElasticPlaneStrain2DLaw").Clone();
+    p_Properties->SetValue(CONSTITUTIVE_LAW,p_constitutive_law);
+    //Create Nodes
+    auto pnode1 = model_part.CreateNewNode(1, 0.00,0.00,0.00);
+    auto pnode2 = model_part.CreateNewNode(2, 1.00,0.00,0.00);
+    auto pnode3 = model_part.CreateNewNode(3, 1.00,1.00,0.00);
+    auto pnode4 = model_part.CreateNewNode(4, 0.00,1.00,0.00);
+    auto pnode5 = model_part.CreateNewNode(5, 0.50,0.00,0.00);
+    auto pnode6 = model_part.CreateNewNode(6, 1.00,0.50,0.00);
+    auto pnode7 = model_part.CreateNewNode(7, 0.50,1.00,0.00);
+    auto pnode8 = model_part.CreateNewNode(8, 0.50,0.50,0.00);
+    //Create Elements
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 1, {{1, 5, 7, 4}}, p_Properties);
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 2, {{5, 2, 6, 8}}, p_Properties);
+    model_part.CreateNewElement("SmallDisplacementElement2D4N", 3, {{8, 6, 3, 7}}, p_Properties);
+    //Add DOF
+    VariableUtils().AddDofWithReaction(DISPLACEMENT_X, REACTION_X, model_part);
+    VariableUtils().AddDofWithReaction(DISPLACEMENT_Y, REACTION_Y, model_part);
+    // Initialize elements
+    auto& r_process_info = model_part.GetProcessInfo();
+    for (auto& elem : model_part.Elements())
```

Comment:
I don't think this is needed since you call `Solve` on the Strategy, can you please test try if it works without?